### PR TITLE
Provide embedding manager

### PR DESCRIPTION
**Pull request by trducng**
image

### CODE REVIEW

File: libs/ktem/flowsettings.py [hunk 1]
```diff
@@ -32,7 +32,15 @@
...
-KH_EMBEDDINGS = {}
+KH_EMBEDDINGS = {
+    "local-mxbai-large-v1": {
+        "spec": {
+            "__type__": "kotaemon.embeddings.FastEmbedEmbeddings",
+            "model_name": "mixedbread-ai/mxbai-embed-large-v1",
+        },
+        "default": True,
+    }
+}
```
Comment (lone17) [inline libs/ktem/flowsettings.py]: currently models can still be configured using the .env file, so could you move this to the end after checking  `KH_EMBEDDINGS ` being empty ?

**Review by lone17 — CHANGES_REQUESTED** [review]

File: libs/ktem/flowsettings.py (hunk 1 above)
Comment (trducng) [reply libs/ktem/flowsettings.py]: I just moved the local declaration into below. Please check @lone17

**Review by lone17 — CHANGES_REQUESTED** [review]
I meant 
```python
    KH_EMBEDDINGS["local"] = {
        "spec": {
            "__type__": "kotaemon.embeddings.EndpointEmbeddings",
            "endpoint_url": "http://localhost:31415/v1/embeddings",
        },
        "default": False,
        "cost": 0,
    }
if len(KH_EMBEDDINGS) < 1:
    KH_EMBEDDINGS["local-mxbai-large-v1"] = {
        "spec": {
            "__type__": "kotaemon.embeddings.FastEmbedEmbeddings",
            "model_name": "mixedbread-ai/mxbai-embed-large-v1",
        },
        "default": True,
    }
```

we add all preconfigured models into the pool, then only set up a default one if no models were set up.

**Comment by trducng** [comment]
But this model is local, shouldn't it reside in the `LOCAL_MODEL`?

**Comment by lone17** [comment]
> But this model is local, shouldn't it reside in the `LOCAL_MODEL`?

the  `LOCAL_MODEL` is a path I used to set up the local endpoint model. not necessarily a flag for using local models. Since your model is small and doesn't need external setting up, I don't think we need a flag for that anyway.

**Review by lone17 — APPROVED** [review]
LGTM

**Pull request merged**